src/scan/postgres_scan.cpp

```diff
@@ -7,12 +9,65 @@
 #include "pgduckdb/pgduckdb_process_lock.hpp"
 #include "pgduckdb/logger.hpp"
 
+#include <numeric> // std::accumulate
+
 namespace pgduckdb {
 
 //
 // PostgresScanGlobalState
 //
 
+static duckdb::string
+FilterJoin(duckdb::vector<duckdb::string> &filters, duckdb::string &&delimiter) {
+	return std::accumulate(filters.begin() + 1, filters.end(), filters[0],
+	                       [&delimiter](duckdb::string l, duckdb::string r) { return l + delimiter + r; });
+}
+
+int
+PostgresScanGlobalState::ExtractQueryFilters(duckdb::TableFilter *filter, const char *column_name,
+                                             duckdb::string &query_filters) {
+	switch (filter->filter_type) {
+	case duckdb::TableFilterType::CONSTANT_COMPARISON:
+	case duckdb::TableFilterType::IS_NULL:
+	case duckdb::TableFilterType::IS_NOT_NULL:
+	case duckdb::TableFilterType::IN_FILTER: {
+		query_filters += filter->ToString(column_name).c_str();
+		return 1;
+	}
+	case duckdb::TableFilterType::CONJUNCTION_OR:
+	case duckdb::TableFilterType::CONJUNCTION_AND: {
+		auto conjuction_filter = reinterpret_cast<duckdb::ConjunctionFilter *>(filter);
+		duckdb::vector<std::string> conjuction_child_filters;
+		for (idx_t i = 0; i < conjuction_filter->child_filters.size(); i++) {
+			std::string child_filter;
+			if (ExtractQueryFilters(conjuction_filter->child_filters[i].get(), column_name, child_filter)) {
+				conjuction_child_filters.emplace_back(child_filter);
+			}
+		}
+		duckdb::string conjuction_delimiter =
+		    filter->filter_type == duckdb::TableFilterType::CONJUNCTION_OR ? " OR " : " AND ";
+		if (conjuction_child_filters.size()) {
+			query_filters += "(" + FilterJoin(conjuction_child_filters, std::move(conjuction_delimiter)) + ")";
+		}
+		return conjuction_child_filters.size();
+	}
+	case duckdb::TableFilterType::OPTIONAL_FILTER: {
+		auto optional_filter = reinterpret_cast<duckdb::OptionalFilter *>(filter);
+		return ExtractQueryFilters(optional_filter->child_filter.get(), column_name, query_filters);
+	}
+	/* DYNAMIC_FILTER is push down filter from topN execution */
+	case duckdb::TableFilterType::DYNAMIC_FILTER:
+		return 0;
```

Comment:
Why do we return 0 here instead of throwing an error? Do we have a query that triggers this?

Reply:
DYNAMIC_QUERY happens when there is `LIMIT` in query so we'll see this filter pushed down to scan, but we can't do anything about it except to skip it.

Reply:
Alright, I looked a bit closer at the code and understand much better what all these things are: I think a somewhat cleaner/future-proof way of handling this is by remembering if we're in an optional filter or not (probably by passing it as a third argument to the recursive function). Then for all these filters that we don't support, we do nothing if it's inside an optional filter, or throw an error if it's not inside an optional filter.